### PR TITLE
Make `truncate_version` return NUMERIC

### PR DESCRIPTION
**Pull request by wlach**
This should work reliably for all the cases we care about and
allows doing things like sorting by versions reliably (since one
can't assume lexicographical ordering).

Using NUMERIC should protect us from weird floating point edge cases:

https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types

See also #1770

### CODE REVIEW

File: sql/mozfun/norm/truncate_version/udf.sql [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
 -- Tests
 SELECT
-  assert.equals("16.1", norm.truncate_version("16.1.1", "minor")),
-  assert.equals("16", norm.truncate_version("16.1.1", "major")),
-  assert.equals("10", norm.truncate_version("10", "minor")),
-  assert.equals("5.1", norm.truncate_version("5.1.5-ubuntu-foobar", "minor")),
-  assert.equals(CAST(NULL AS STRING), norm.truncate_version("5.1.5-ubuntu-foobar", "patch")),
-  assert.equals(CAST(NULL AS STRING), norm.truncate_version("foo-bar", "minor"))
+  assert.equals(16.1, norm.truncate_version("16.1.1", "minor")),
+  assert.equals(16.03, norm.truncate_version("16.03.1", "minor")),
```
Comment (jklukas) [inline sql/mozfun/norm/truncate_version/udf.sql:20]: This case actually worries me a bit. Do we see minor versions like "03" in the wild? I could imagine a minor version "2" without the zero padding that should sort before "03" logically, but the numeric value we return is `16.03`.

I think I am worrying too much, and this is probably fine.

**Review by jklukas — APPROVED** [review]
r+wc

I think the docs for this function should be updated to explicitly discuss the return type and the benefits of NUMERIC for sorting.

File: sql/mozfun/norm/truncate_version/udf.sql (hunk 1 above)
Comment (wlach) [reply sql/mozfun/norm/truncate_version/udf.sql:20]: Ubuntu numbers things like "16.04" (YY.MM), so it's not unheard of. But I don't think they mix/match versioning like you suggest for your pathological case: the next version after "16.04" was "16.10" IIRC.

**Comment by wlach** [comment]
@jklukas quick sanity check on the doc updates?

**Review by jklukas — APPROVED** [review]

**Pull request merged**